src/app/component/general/eco-news/eco-news-detail/eco-news-widget/eco-news-widget.component.ts

```diff
@@ -14,25 +14,30 @@ import { Subscription } from 'rxjs';
 export class EcoNewsWidgetComponent implements OnInit, OnDestroy {
   @Output() ecoNewsModel: EcoNewsModel;
   @Output() idNumber = new EventEmitter<string>();
+  private lastThreeNewsSubscription: Subscription;
+  private idNewsGotSubscription: Subscription;
+  private SortedNews: EcoNewsModel[];
+  private selectedId: number;
   private profileIcons = ecoNewsIcons;
   private defaultPicture = ecoNewsIcons;
-  private idNumberFromUrl: number;
-  private SortedNews: EcoNewsModel[];
-  private lastThreeNewsSubscription: Subscription;
 
-  constructor(
-    private ecoNewsService: EcoNewsService,
-    private route: ActivatedRoute
-    ) { }
+  constructor(private ecoNewsService: EcoNewsService,
+              private route: ActivatedRoute) { }
 
   ngOnInit() {
-    this.fetchSortedNews();
-    this.bindNewsToModel();
+    this.newsIdSubscription();
+  }
+
+  private newsIdSubscription(): void {
+    this.idNewsGotSubscription = this.route.paramMap.subscribe(param => {
+      this.selectedId = +param.get('id');
+      this.fetchSortedNews();
+      this.bindNewsToModel();
+    });
   }
 
-  private fetchSortedNews(): void {
-    this.ecoNewsService.selectedId = +this.route.snapshot.paramMap.get('id');
-    this.ecoNewsService.sortLastThreeNewsChronologically();
+  private fetchSortedNews(): any {
```

Comment:
please change return type to the appropriate

Reply:
fixed